hack/prowimagebuilder/main.go

```diff
@@ -253,7 +272,7 @@ func main() {
 			for {
 				select {
 				case id := <-imageChan:
-					if err := buildAndPush(&id, o.koDockerRepo, o.push); err != nil {
+					if err := buildAndPush(&id, o.koDockerRepo, o.push, imageChan, &wg); err != nil {
```

Comment:
In my opinion it would simpler to implement the retry logic as a loop here rather than adding requeuing to every return statement and passing down the WaitGroup and input channel. This looks correct too though.

Reply:
excellent idea, let me update

Reply:
done